src/consensus/params.h

```diff
@@ -91,7 +93,7 @@ struct Params {
     NetworkUpgrade vUpgrades[MAX_NETWORK_UPGRADES];
     /** Proof of work parameters */
     uint256 powLimit;
-    bool fPowAllowMinDifficultyBlocks;
+    boost::optional<uint32_t> nPowAllowMinDifficultyBlocksFromHeight;
```

Comment:
The name of this field has an off-by-one error relative to its semantics. The name would suggest that the block at this height, and subsequent blocks, are allowed to be min-difficulty. In fact only subsequent blocks are allowed to be min-difficulty. Renaming it to ``nPowAllowMinDifficultyBlocksAfterHeight`` would resolve this discrepancy.

Reply:
I agree.  For example 299187 is the current testnet block height, and what we are saying is we can mine min difficulty blocks after 299187.

Reply:
Or set above to 299188.